plugin/pkg/shoot/validator/admission.go

```diff
@@ -655,20 +655,20 @@ func (c *validationContext) validateShootNetworks(workerless bool) field.ErrorLi
 				c.shoot.Spec.Networking.Services,
 				workerless,
 			)...)
-		}
 
-		// validate network disjointedness with seed networks if shoot is being (re)scheduled
-		if !apiequality.Semantic.DeepEqual(c.oldShoot.Spec.SeedName, c.shoot.Spec.SeedName) {
-			allErrs = append(allErrs, cidrvalidation.ValidateNetworkDisjointedness(
-				path,
-				c.shoot.Spec.Networking.Nodes,
-				c.shoot.Spec.Networking.Pods,
-				c.shoot.Spec.Networking.Services,
-				c.seed.Spec.Networks.Nodes,
-				c.seed.Spec.Networks.Pods,
-				c.seed.Spec.Networks.Services,
-				workerless,
-			)...)
+			// validate network disjointedness with seed networks if shoot is being (re)scheduled
+			if !apiequality.Semantic.DeepEqual(c.oldShoot.Spec.SeedName, c.shoot.Spec.SeedName) {
```

Comment:
Does seems to work in my local setup.

On DELETE request, `c.shoot` does not have an `deletionTimestamp`. See
```
{"level":"info","ts":"2023-06-21T10:37:00.883Z","msg":"shoot - &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:local GenerateName: Namespace:garden-local SelfLink: UID:389acda0-f293-4dc7-8cd8-7f06f5342a0d ResourceVersion:745 Generation:1 CreationTimestamp:2023-06-21 08:55:30 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> ...
```

Hence it enters in the above if statement (c.shoot.DeletionTimestamp == nil).

The deletion of such affected Shoot still fails with:
```
% k -n garden-local delete shoot local
Error from server (Forbidden): shoots.core.gardener.cloud "local" is forbidden: [spec.networking.services: Invalid value: "192.168.0.0/17": shoot service network intersects with default vpn network (192.168.123.0/24)]
```

Reply:
Then let's check `if a.GetOperation() == admission.Delete`

Reply:
Thanks a lot for catching this.
I addressed the issue in https://github.com/gardener/gardener/pull/8131.